Teleport cloud license info and other updates (#11093)
gravitational/teleport: docs/pages/setup/reference/license.mdx

description: Teleport enterprise license file configuration parameters and requirements
---

Commercial self-hosted Teleport subscriptions require a valid license. The license file can
be downloaded from the [Teleport Customer

> Portal](https://dashboard.gravitational.com/web/login). When downloading the license file the licensed products for the Teleport Customer will list within the download screen. 

Reviewer: ```suggestion
Portal](https://dashboard.gravitational.com/web/login). When downloading the file, the products you are licensed to use will be listed in the browser. 
```
Made this suggestion to remove some repetition of "license" and be a bit more specific about what "download screen" means. Is "the browser" correct here?

> Portal](https://dashboard.gravitational.com/web/login). When downloading the license file the licensed products for the Teleport Customer will list within the download screen. 

Reviewer 2: Maybe "the portal" would be more clear?

> <Admonition type="note" title="Not applicable to Teleport Cloud">

Reviewer: I think we should make this a `Notice` component with `scope={["cloud"}` so only Cloud customers see this.

Author: I would leave it in since this a common confusion.  They may also be researching their setup and the enterprise section applies to cloud and enterprise users.

Reviewer: That makes sense—what would you think of a `Notice` without the `scope` and with `type="tip`"? I think that would be cleaner than the current Admonition, since the title and body text mainly repeat one another.

> <Admonition type="note" title="Not applicable to Teleport Cloud">

Reviewer: I think we should make this a `Notice` component with `scope={["cloud"}` so only Cloud customers see this.

Author: See above comment. Enterprise customers also switch to cloud so good for them to know this will no longer apply.

> Teleport Cloud manages the Teleport licensing for customers.  No license file management is required in Teleport Cloud to manage by a Teleport Customer.

Reviewer: ```suggestion
Teleport Cloud manages licensing for customers, and it is no need to manage license files yourself. 
```
to trim this a bit

`data_dir`. If the license file path is not set, Teleport will look for the
`license.pem` file in the configured `data_dir`.

> The Teleport l

Reviewer: Just flagging this to either remove or complete

> The Teleport l

Reviewer: Just flagging this to either remove or complete

<Admonition
  type="tip"
  title="NOTE"
>

>   Only Auth Servers require the license. Proxies and Nodes that do not also have Auth role enabled do not need the license.  

Reviewer: Rather than expand this Admonition box, I think we can remove the `Admonition` tags and make it regular text instead. Since it's a short page, I don't think it will be very easy to skip a paragraph.

>   Only Auth Servers require the license. Proxies and Nodes that do not also have Auth role enabled do not need the license.  

Reviewer: Rather than expand this Admonition box, I think we can remove the `Admonition` tags and make it regular text instead. Since it's a short page, I don't think it will be very easy to skip a paragraph.

    
    Attempts to use unlicenced products will result in a error message and users will be unable to use those products.  Here is an example message if attempting to use database access and not included in the license.

> ```bash

Reviewer: The coloration is a bit off here. Maybe use `code` instead of `bash`? Or is there a `text` type?

> ```bash

Reviewer: The coloration is a bit off here. Maybe use `code` instead of `bash`? Or is there a `text` type?

`data_dir`. If the license file path is not set, Teleport will look for the
`license.pem` file in the configured `data_dir`.

Only Auth Servers require the license. Proxies and Nodes that do not also have Auth role enabled do not need the license.  
    

> Attempts to use unlicenced products will result in a error message and users will be unable to use those products.  Here is an example message if attempting to use database access and not included in the license.

Author: done

Only Auth Servers require the license. Proxies and Nodes that do not also have Auth role enabled do not need the license.  
    

> Attempts to use unlicensed products will result in a error message and users will be unable to use those products.  or example, if you attempt to use database access without it being included in the license, you will get a message similar to:

Reviewer 2: Just one missing F in For :)

Author: fixed!
